src/Features/CSharp/Portable/Completion/CompletionProviders/ObjectAndWithInitializerCompletionProvider.cs

```diff
@@ -143,20 +143,17 @@ internal override bool IsInsertionTrigger(SourceText text, int characterPosition
 
         private static ITypeSymbol GetInitializedType(SyntaxToken token, Document document, SemanticModel semanticModel, CancellationToken cancellationToken)
         {
-            // new Goo { bar = $$
-            if (token.Parent.Parent.IsKind(SyntaxKind.ObjectCreationExpression, out ObjectCreationExpressionSyntax objectCreation))
-            {
-                return semanticModel.GetSymbolInfo(objectCreation.Type, cancellationToken).Symbol as ITypeSymbol;
-            }
+            var parent = token.Parent.Parent;
 
-            // new() { bar = $$
-            if (token.Parent.Parent.IsKind(SyntaxKind.ImplicitObjectCreationExpression, out ImplicitObjectCreationExpressionSyntax implicitObjectCreation))
+            // new() { $$
+            // new Goo { $$
+            if (parent.IsKind(SyntaxKind.ObjectCreationExpression, SyntaxKind.ImplicitObjectCreationExpression))
             {
-                return semanticModel.GetTypeInfo(implicitObjectCreation, cancellationToken).ConvertedType;
```

Comment:
ConvertedType will get to the nullable type rather than the type itself causing the wrong members to show up.